quic/s2n-quic-transport/src/endpoint/mod.rs

```diff
@@ -555,7 +556,22 @@ impl<Cfg: Config> Endpoint<Cfg> {
             .lookup_internal_connection_id(&destination_connection_id)
         {
             let mut check_for_stateless_reset = false;
-            let mtu_config = self.mtu_config;
+            let info = mtu::ConnectionInfo::new(&remote_address, self.endpoint_mtu_config);
+            let mtu_config =
+                match CheckedConfig::new(endpoint_context.mtu.on_connection(&info), &info) {
+                    Ok(config) => config,
+                    Err(_) => {
+                        publisher.on_endpoint_datagram_dropped(
+                            event::builder::EndpointDatagramDropped {
+                                len: payload_len as u16,
+                                reason: event::builder::DatagramDropReason::MtuValidation {
+                                    endpoint_mtu_config: self.endpoint_mtu_config.into_event(),
+                                },
+                            },
+                        );
+                        return;
+                    }
+                };
```

Comment:
so the only reason this is here is to handle connection migration, since at this point we've already established a connection and validated the mtu config that was created based on the original remote address. 

I think it probably does not make sense to call the `mtu` provider on every single datagram, but it does probably make sense to support different MTU config for a migrated path to handle a client migrating from wi-fi to 5G.

So you could either wire the `mtu` provider (alongside the endpoint mtu config) down through to the `path::Manager`, which would only invoke it if it was actually a connection migration, or somehow determine it is a connection migration at this level (probably by comparing remote addresses). Might need a second opinion on that @camshaft

Reply:
Yeah we definitely should only call it on a new valid path.

And that actually brings up a good point: the method should probably be called `on_path`, not `on_connection`. And probablye not `ConnectionInfo`, but `PathInfo` instead.